src/components/color-picker/color-picker.tsx

```diff
@@ -1087,21 +1090,25 @@ export class ColorPicker implements InteractiveComponent {
 
   async fetchStrings(): Promise<void> {
     // TODO: these could split into reusable utils
+    // setup util for fetching locale
+
+    //fetch supported locales
     const locale = this.getLocale();
     // TODO: fetching could be moved to a util (similar to icon fetching)
-    this.builtInStrings = await (
-      await fetch(getAssetPath(`./assets/color-picker/i18n/${locale}.json`))
-    ).json();
+    this.builtInStrings = await fetchLocaleStrings(locale, "color-picker");
   }
 
   getLocale(): string {
-    const lang = document.documentElement.lang || navigator.language; // need to determine best place to grab locale from
+    // need to determine best place to grab locale from
+    //we can go by order ( locale of component > locale of closests > lang of document > navigator)
+    const lang = document.documentElement.lang || navigator.language;
+
     return this.normalizeLocale(lang);
   }
 
-  normalizeLocale(locale: string): string {
+  normalizeLocale(lang: string): string {
```

Comment:
This and `getLocale` were just placeholder methods. They can be dropped in favor of the string fetching util.